### PR TITLE
[SPARK-21606][SQL]HiveThriftServer2 catches OOMs on request threads

### PR DESCRIPTION
**Pull request by zuotingbing**
## What changes were proposed in this pull request?

Refer to Hive, ThriftCLIService methods such as ExecuteStatement are apparently capable of catching OOMs because they get wrapped in RTE by HiveSessionProxy.This shouldn't happen.
I create a PR to fix this bug in Spark also.

## How was this patch tested?

Exist tests and manual tests

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @liufengdb Is this PR in the right track?

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java [hunk 1]
```diff
@@ -37,21 +30,29 @@
 import org.apache.thrift.transport.TServerSocket;
 import org.apache.thrift.transport.TTransportFactory;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
```
Comment (liufengdb) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java]: unnecessary change

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java [hunk 2]
```diff
@@ -39,6 +32,8 @@
 import org.apache.hive.service.cli.thrift.ThriftCLIService;
 import org.apache.hive.service.cli.thrift.ThriftHttpCLIService;
 
+import java.util.Properties;
```
Comment (liufengdb) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/server/HiveServer2.java]: java imports should be the first?

**Comment by liufengdb** [comment]
I actually did not get the motivation of this PR. HiveThriftServer2 can run independently or be started with a SQL context: https://github.com/apache/spark/pull/18812/files#diff-709404b0d3defeff035ef0c4f5a960e5L57. For the later, this OOM should be handled by the upper layer, right?

**Comment by HyukjinKwon** [comment]
gentle ping @zuotingbing
